cri/v1alpha2/cri_stream.go

```diff
@@ -0,0 +1,159 @@
+package v1alpha2
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"os/exec"
+	"strings"
+	"time"
+
+	apitypes "github.com/alibaba/pouch/apis/types"
+	"github.com/alibaba/pouch/cri/stream/remotecommand"
+	"github.com/alibaba/pouch/daemon/mgr"
+
+	"github.com/sirupsen/logrus"
+)
+
+func newStreamServer(ctrMgr mgr.ContainerMgr, address string, port string) (Server, error) {
+	config := DefaultConfig
+	config.Address = net.JoinHostPort(address, port)
+	runtime := newStreamRuntime(ctrMgr)
+	return NewServer(config, runtime)
+}
+
+type streamRuntime struct {
+	containerMgr mgr.ContainerMgr
+}
+
+func newStreamRuntime(ctrMgr mgr.ContainerMgr) Runtime {
+	return &streamRuntime{containerMgr: ctrMgr}
+}
+
+// Exec executes a command inside the container.
+func (s *streamRuntime) Exec(containerID string, cmd []string, streamOpts *remotecommand.Options, streams *remotecommand.Streams) (uint32, error) {
+	createConfig := &apitypes.ExecCreateConfig{
+		Cmd:          cmd,
+		AttachStdin:  streamOpts.Stdin,
+		AttachStdout: streamOpts.Stdout,
+		AttachStderr: streamOpts.Stderr,
+		Tty:          streamOpts.TTY,
+	}
+
+	ctx := context.Background()
+
+	execid, err := s.containerMgr.CreateExec(ctx, containerID, createConfig)
+	if err != nil {
+		return 0, fmt.Errorf("failed to create exec for container %q: %v", containerID, err)
+	}
+
+	startConfig := &apitypes.ExecStartConfig{}
+	attachConfig := &mgr.AttachConfig{
+		Streams:     streams,
+		MuxDisabled: true,
+	}
+
+	err = s.containerMgr.StartExec(ctx, execid, startConfig, attachConfig)
+	if err != nil {
+		return 0, fmt.Errorf("failed to start exec for container %q: %v", containerID, err)
+	}
+
```

Comment:
Please add the todo/fixme as reminder. I think we can find the better way to handle this case gracefully instead of the dead loop. WDYT?

Reply:
@fuweid  All mentioned above has been done.